docs/src/main/asciidoc/_admin-guide/subsystem-configuration/EJB3.adoc

```diff
@@ -174,6 +184,51 @@ This is used to enable remote EJB invocations. It specifies a space-separated li
 connectors to use (as defined in the remoting subsystem configuration),
 and the thread pool to use for remote invocations.
 
+[[profile]]
+=== <profile>
+
+Remote profile specificies a configuration of remote invocations that can
+be referenced by many deployments. EJBs that are meant to be invoked can
+be discovered in either static or dynamic way.
+
+Static discovery decides which remote node to connect based on the information
+provided by the user.
+
+Dynamic discovery is responsible for monitoring available EJBs on all the
+nodes to which connections are configured and decides which remote node to
+connect based on gathered data.
+
+[[static-ejb-discovery]]
+==== <static-ejb-discovery>
+
+Static ejb discovery allows to explicitly specify on which remote nodes
+given EJBs are located. `module` tag is used to define it:
+
+* `module-name` the name of EJB module
+* `app-name` the name of EJB app
+* `distinct-name` the distinct name EJB
+* `uri` the address on which given EJB is located
+
+[[remoting-ejb-receiver]]
+==== <remoting-ejb-receiver>
+
+`remoting-ejb-receiver` tag is used to define dynamic discovery based on
+the remoting procol:
+
+* `name` name of the remote connection
+* `outbound-connection-ref` reference to outbound connection defined in
+the remoting subsystem
+* `connection-timeout` the timeout of the connection
+
+[[remote-http-connection]]
+==== <remote-http-connection>
+
+`remote-http-connection` tag is used to define dynamic discovery based on
+HTTP procol:
```

Comment:
s/procol/protocol

Reply:
fixed

Reply:
I think the main question I would like to ask is - is there sufficient test coverage of clients using authentication client configuration for a dynamically discovered destination?

As mentioned earlier we have aspects such as SSL and the credentials for authentication which should already be covered by the EJB client to verification that the behaviour is correct.  In this case I believe it will be preferable for the resolved configuration to be based on the discovered destination.

Reply:
The test coverage for the dynamically discovered destinations would require additional analysis. OTOH, dynamic discovery is main route of invocation also for remoting and this commit just adds this feature for HTTP. Currently EAP/WildFly already relies on dynamic invocation and if we decide that the testsuite have to be extended it should be for done a number of scenarios, possibly independently of this PR.

Reply:
Ok, lets go with what we have here - if you are happy HTTP is just using the same pattern as the existing approaches we should be good.

Reply:
Hmm, but just to clarify cause I'm not sure if I understand correctly. Discovery provider and HttpEjbReceiver (which is used to perform invocation to discovered destination) (also other receivers) resolve their AuthenticationContexts independently:
https://github.com/wildfly/wildfly-http-client/blob/master/ejb/src/main/java/org/wildfly/httpclient/ejb/HttpEJBReceiver.java#L132

Should this be modified?